niworkflows/utils/bids.py

```diff
@@ -315,6 +304,46 @@ def _grp_echos(x):
     return ses_uids
 
 
+def relative_to_root(path):
+    """
+    Calculate the BIDS root folder given one file path's.
+
+    Example
+    -------
+    >>> str(relative_to_root(
+    ...     "/sub-03/sourcedata/sub-01/anat/sub-01_T1.nii.gz"
+    ... ))
+    'sub-01/anat/sub-01_T1.nii.gz'
+
+    >>> str(relative_to_root(
+    ...     "/sub-03/anat/sourcedata/sub-01/ses-preop/anat/sub-01_ses-preop_T1.nii.gz"
+    ... ))
+    'sub-01/ses-preop/anat/sub-01_ses-preop_T1.nii.gz'
+
+    >>> str(relative_to_root(
+    ...     "sub-01/anat/sub-01_T1.nii.gz"
+    ... ))
+    'sub-01/anat/sub-01_T1.nii.gz'
+
+    >>> str(relative_to_root("anat/sub-01_T1.nii.gz"))
+    'anat/sub-01_T1.nii.gz'
+
+    """
+    path = Path(path)
+    if path.name.startswith("sub-"):
+        parents = [path.name]
+        for p in path.parents:
+            parents.insert(0, p.name)
+            if p.name.startswith("sub-"):
+                return Path(*parents)
+        return path
+
+    raise NotImplementedError(
```

Comment:
`ValueError` seems more appropriate.

```suggestion
    raise ValueError(
```